src/libtiled/templatemanager.cpp

```diff
@@ -45,13 +45,17 @@ void TemplateManager::deleteInstance()
 }
 
 TemplateManager::TemplateManager(QObject *parent)
-    : QObject(parent)
+    : QObject(parent),
+      mWatcher(new FileSystemWatcher(this))
 {
+    connect(mWatcher, &FileSystemWatcher::fileChanged,
+            this, &TemplateManager::fileChanged);
 }
 
 TemplateManager::~TemplateManager()
 {
     qDeleteAll(mObjectTemplates);
+    delete mWatcher;
```

Comment:
There's no need to delete the watcher explicitly, because you've passed the `TemplateManager` as its parent in the constructor.

Reply:
I'm still learing the Qt memory rules and such. I'll remove that.